test/controllers/projects/samples/metadata/fields/fields_controller_test.rb

```diff
@@ -0,0 +1,86 @@
+                    'value' => { 'value1' => 'value1' }
+                  } }, format: :turbo_stream }
+            assert_response :ok
+          end
+
+          test 'update metadata value' do
+            patch namespace_project_sample_metadata_field_path(@namespace, @project29, @sample32),
+                  params: { 'sample' => { 'update_field' => {
+                    'key' => { 'metadatafield1' => 'metadatafield1' },
+                    'value' => { 'value1' => 'value2' }
+                  } }, format: :turbo_stream }
+            assert_response :ok
+          end
+
+          test 'update metadata key and value' do
+            patch namespace_project_sample_metadata_field_path(@namespace, @project29, @sample32),
+                  params: { 'sample' => { 'update_field' => {
+                    'key' => { 'metadatafield1' => 'metadatafield3' },
+                    'value' => { 'value1' => 'value2' }
+                  } }, format: :turbo_stream }
+            assert_response :ok
+          end
+
+          test 'cannot update metadata key with key that already exists' do
+            patch namespace_project_sample_metadata_field_path(@namespace, @project29, @sample32),
+                  params: { 'sample' => { 'update_field' => {
+                    'key' => { 'metadatafield1' => 'metadatafield2' },
+                    'value' => { 'value1' => 'value1' }
+                  } }, format: :turbo_stream }
+            assert_response :unprocessable_entity
+          end
+
+          test 'update metadata with unchanged metadata' do
+            patch namespace_project_sample_metadata_field_path(@namespace, @project29, @sample32),
+                  params: { 'sample' => { 'update_field' => {
+                    'key' => { 'metadatafield1' => 'metadatafield1' },
+                    'value' => { 'value1' => 'value1' }
+                  } }, format: :turbo_stream }
+            assert_response :unprocessable_entity
+          end
+
+          test 'cannot update metadata if sample does not belong to the project' do
+            patch namespace_project_sample_metadata_field_path(@namespace, @project4, @sample32),
+                  params: { 'sample' => { 'update_field' => {
+                    'key' => { 'metadatafield1' => 'metadatafield3' },
+                    'value' => { 'value1' => 'value3' }
+                  } }, format: :turbo_stream }
+            assert_response :not_found
+          end
+
+          test 'cannot update sample if not a member with access to the project' do
+            sign_in users(:david_doe)
+            patch namespace_project_sample_metadata_field_path(@namespace, @project29, @sample32),
+                  params: { 'sample' => { 'update_field' => {
+                    'key' => { 'metadatafield1' => 'metadatafield3' },
+                    'value' => { 'value1' => 'value3' }
+                  } }, format: :turbo_stream }
+            assert_response :unauthorized
+          end
+        end
```

Comment:
Please add a test for a user not being able to update analysis provided metadata

Reply:
Updated in 2a79e5b24b98f16b80cf676792daa31801249850